src/parser/mod.rs

```diff
@@ -147,17 +273,26 @@ pub(self) fn has_group(replacement: &str) -> bool {
 pub(self) fn replace_cow<'a>(
     replacement: &str,
     replacement_has_group: bool,
-    captures: &regex::Captures,
+    captures: &regex::bytes::Captures,
 ) -> Cow<'a, str> {
     if replacement_has_group && captures.len() > 0 {
-        let mut target = String::with_capacity(31);
-        captures.expand(replacement, &mut target);
-        Cow::Owned(target.trim().to_owned())
+        let mut target = vec![];
+        let raw_replacement = replacement.as_bytes();
+        captures.expand(raw_replacement, &mut target);
+        std::str::from_utf8(&target)
+            .map(|s| Cow::Owned(s.trim().to_owned()))
+            // What is the behavior if we can't parse a string???
```

Comment:
This is the only big question mark in behavior. If a replacement has a group and the result of the capture expands into a `&[u8]` that can't be turned into a string, then we have to do something. Rather than panic or try and surface this, I chose to just skip the replacement.

This may never happen. It likely won't. I'm not sure if a panic is better or worse here and am very open to feedback.